client/components/dot-pager/index.jsx

```diff
@@ -128,31 +163,116 @@ export const DotPager = ( {
+			return { x: event.targetTouches[ 0 ].clientX, y: event.targetTouches[ 0 ].clientY };
+		}
+
+		const touch = event.changedTouches[ 0 ];
+		return { x: touch.clientX, y: touch.clientY };
+	};
+
+	const handleDragStart = ( event ) => {
+		const position = getDragPosition( event );
+		setDragStartData( position );
+		setPagesStyle( { ...pagesStyle, transitionDuration: `0ms` } ); // Set transition Duration to 0 for smooth dragging.
+		setIsDragging( true );
+	};
+
+	const handleDrag = ( event ) => {
+		if ( ! isDragging ) {
+			return;
+		}
+		const dragPosition = getDragPosition( event );
+		const delta = dragPosition.x - dragStartData.x;
+
+		const offset = getOffset( currentPage ) + delta;
+		// Allow for swipe left
+		if ( numPages !== currentPage + 1 && delta < 0 ) {
+			setPagesStyle( {
+				...pagesStyle,
+				transform: `translate3d(${ offset }px, 0px, 0px)`,
+			} );
+		} else if ( currentPage !== 0 && delta > 0 ) {
+			setPagesStyle( {
+				...pagesStyle,
+				transform: `translate3d(${ offset }px, 0px, 0px)`,
+			} );
+		}
+	};
+
+	const handleDragEnd = ( event ) => {
+		const THRESHOLD_OFFSET = 100; // Number of pixels to travel before we trigger the slider to move to the desired slide.
+		const dragPosition = getDragPosition( event );
+		const delta = dragPosition.x - dragStartData.x;
+
+		const hasMetThreshold = Math.abs( delta ) > THRESHOLD_OFFSET;
+
+		let newIndex = currentPage;
+		if ( delta < 0 && hasMetThreshold && numPages !== currentPage + 1 ) {
+			newIndex = currentPage + 1;
+		}
+
+		if ( delta > 0 && hasMetThreshold && currentPage !== 0 ) {
+			newIndex = currentPage - 1;
+		}
+		const offset = getOffset( newIndex );
+		setPagesStyle( {
+			transform: `translate3d(${ offset }px, 0px, 0px)`,
+		} );
+		setCurrentPage( newIndex );
+		setDragStartData( {} );
+		setIsDragging( false );
+	};
+
 	return (
-		<div className={ className } { ...props }>
+		<div className={ classnames( 'dot-pager', className ) } { ...props }>
 			<Controls
 				showControlLabels={ showControlLabels }
 				currentPage={ currentPage }
 				numberOfPages={ numPages }
 				setCurrentPage={ ( index ) => {
 					onPageSelected && onPageSelected( index );
+					const offset = getOffset( index );
+					setPagesStyle( {
+						...pagesStyle,
+						transform: `translate3d(${ offset }px, 0px, 0px)`,
+						transitionDuration: `300ms`,
+					} );
 					setCurrentPage( index );
 				} }
 			/>
-			<div className="dot-pager__pages" ref={ pagesRef } style={ pagesStyle }>
-				{ Children.map( children, ( child, index ) => (
-					<div
-						className={ classnames( 'dot-pager__page', {
-							'is-current': index === currentPage,
-							'is-prev': index < currentPage,
-							'is-next': index > currentPage,
-						} ) }
-						key={ `page-${ index }` }
-					>
-						{ child }
-					</div>
-				) ) }
+			<div
+				className="dot-pager__container"
+				onDragStart={ handleDragStart }
+				onPointerDown={ handleDragStart }
+				onDrag={ handleDrag }
```

Comment:
I don't think it hurts to have it on a desktop as well. It did help when I was developing it. Do you have any concerns? Or just that it might not be intended?

Reply:
Yeah, it could be unintended. Also, I think it should be something we can "plug in" to a component, rather than implementing it so coupled to the dot pager component. I also mentioned that in the review, what do you think:

> That being said, mobile navigation with touch feels like a very generic behavior that we should be able to reuse in multiple areas in the future. Is it possible to implement it in a reusable way, so we will be able to apply it easily to other areas of the UI? Maybe it could be implemented as an independent component?

Also, I think we should loop in some dotcom designers here just in case.

Reply:
> I think we should loop in some dotcom designers here just in case.

cc @olaolusoga  Does it make sense to have this feature on desktop as well or would you rather see it removed?

Reply:
Thanks for ping.

>I think it should be something we can "plug in" to a component, rather than implementing it so coupled to the dot pager component

If there's value in decoupling for ease long term, and scale then agree that we should. Better to do it now, than in the future where it might be more complicated due to broader usage.

>Does it make sense to have this feature on desktop as well or would you rather see it removed?

It doesn't hurt. It's a nice easter egg if a person is using hardware that allows them to access gestures on desktop.